### PR TITLE
Added custom http option

### PR DESCRIPTION
**Pull request by AsafMah**
#### Pull Request Description

Users need to be able to specify/configure http settings like
proxy & custom retries, which are only avaiable to set by exposing
http.Client as something the user can customize. For example,
in enterprise boxes, deployments may not have access to internet
and need to go through squid proxy. Setting HTTPS_PROXY is invasive,
as it forces other http clients to disable proxy in transport or edit NO_PROXY
in such a way that specific dns suffix are disabled. Furthermore, even if HTTPS_PROXY 
is specified, some proxy requires users to specify certs/authorization header.

#### Future Release Comment
package `github.com/azure-kusto-go/kusto` supports `NewWithCustomHttp` allowing users to instantiate
a query Kusto client with http customizations.

**Breaking Changes:**
- None

**Features:**
- This addresses https://github.com/Azure/azure-kusto-go/issues/46 from a code perspective

**Fixes:**
- None

### CODE REVIEW

**Review by yogilad — COMMENTED** [review]
How does this change cover ingestion flows i.e. how does it cover access to SAs?

**Comment by AsafMah** [comment]
This is just for queries, users provide their own Auth, so they'll need to set whatever proxy or setting they have there.

**Comment by AsafMah** [comment]
@yogilad Ok, I also made it so the http client will go to the ingestion flow

File: kusto/ingest/internal/conn/conn.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type Conn struct {
 }
 
 // New returns a new Conn object.
-func New(endpoint string, auth kusto.Authorization) (*Conn, error) {
+func New(endpoint string, auth kusto.Authorization, client *http.Client) (*Conn, error) {
```
Comment (14leeyuchieh) [inline kusto/ingest/internal/conn/conn.go:47]: Should this be called out as a breaking change? Technically, the constructor for the connection changed, right?

Comment (AsafMah) [reply kusto/ingest/internal/conn/conn.go:47]: It's an internal file, so it's not breaking to change it.

**Review by yogilad — APPROVED** [review]

**Pull request merged**